cpu/cc2538/periph/uart.c

```diff
@@ -189,30 +129,12 @@ static int init_base(uart_t uart, uint32_t baudrate)
 {
     cc2538_uart_t *u = NULL;
 
-    switch (uart) {
-#if UART_0_EN
-        case UART_0:
-            u = UART_0_DEV;
-            gpio_init_af(UART_0_RX_PIN, UART0_RXD, GPIO_IN);
-            gpio_init_af(UART_0_TX_PIN, UART0_TXD, GPIO_OUT);
-            break;
-#endif
-#if UART_1_EN
-        case UART_1:
-            u = UART_1_DEV;
-            gpio_init_af(UART_1_RX_PIN, UART1_RXD, GPIO_IN);
-            gpio_init_af(UART_1_TX_PIN, UART1_TXD, GPIO_OUT);
-            break;
-#endif
-
-        default:
-            (void)u;
-            return UART_NODEV;
-    }
+    u = uart_config[uart].dev;
+    gpio_init_af(uart_config[uart].rx_pin, uart_config[uart].rxd, GPIO_IN);
+    gpio_init_af(uart_config[uart].tx_pin, uart_config[uart].txd, GPIO_OUT);
```

Comment:
as stated above, the values for `rxd` and `txd` are static as far as I can see, so no need to have them in the board config.